src/classes/RD2_RecurringDonationsOpp_TDTM.cls

```diff
@@ -56,12 +56,44 @@ public class RD2_RecurringDonationsOpp_TDTM extends TDTM_Runnable {
 
         TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.RD, true);
 
+        findMatchingElevateRecurringDonations(opps, triggerAction);
         evaluateOpportunities(opps, oldOpps, triggerAction);
 
         TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.RD, false);
         return null;
     }
 
+     /**
+    * @description When new opportunities are inserted that have a value in the CommitmentId field
+    * look for matching existing Recurring Donations with that same CommmitmentId.
+    * @param opps Inserted Opportunities
+    * @param triggerAction BeforeInsert, AfterInsert, etc.
+    */
+    private void findMatchingElevateRecurringDonations(List<Opportunity> opps, TDTM_Runnable.Action triggerAction) {
```

Comment:
The Opp evaluation for the RD match should be done in the Elevate service, thus this method can be re-written as:
```
    /**
    * @description Match new Opportunities not referencing any Recurring Donation and originated from Elevate 
    * with a Recurring Donation having the same Commitment Id.
    * @param opps Opportunities to be created
    * @param triggerAction BeforeInsert, AfterInsert, etc.
    * @return void
    */
    private void matchOppsToRecurringDonations(List<Opportunity> opps, TDTM_Runnable.Action triggerAction) {
        if (triggerAction != TDTM_Runnable.Action.BeforeInsert) {
            return;
        }

        new RD2_ElevateIntegrationService()
            .matchOppsToRecurringDonations(opps);
    }
```